WordPress/Sniff.php

```diff
@@ -36,6 +36,15 @@
 abstract class Sniff implements PHPCS_Sniff {
 
 	/**
+	 * Regex to get complex variables from T_DOUBLE_QUOTED_STRING or T_HEREDOC.
+	 *
+	 * @since 0.14.0
+	 *
+	 * @var string
+	 */
+	const REGEX_COMPLEX_VARS = '`(?:(\{)?(?<!\\\\)\$)?(\{)?(?<!\\\\)\$(\{)?(?P<varname>[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*)(?:->\$?(?P>varname)|\[[^\]]+\]|::\$?(?P>varname)|\([^\)]*\))*(?(3)\}|)(?(2)\}|)(?(1)\}|)`';
```

Comment:
Where has this regex come from? 

Can it be broken down and commented (with the flag that ignores whitespace)?

Reply:
I wrote the regex (what did you expect ?).

I did so nearly a year ago when I needed the same `strip_interpolated_variables()` function for the [PHPCompatibility standard](https://github.com/wimg/PHPCompatibility/blob/9a9d15aa50b6ffec7a0e7683377bef338c64911f/PHPCompatibility/Sniff.php#L26) and pulled it in https://github.com/wimg/PHPCompatibility/pull/314 which includes [extensive unit tests of the function](https://github.com/wimg/PHPCompatibility/blob/9a9d15aa50b6ffec7a0e7683377bef338c64911f/PHPCompatibility/Tests/BaseClass/FunctionsTest.php#L441-L504).

The varname part of the regex comes from http://php.net/manual/en/language.variables.basics.php

I don't mind documenting the regex, but that will easily take me two hours or more as I'll have to decipher it myself again first... 

If you insist on it being documented, I'd suggest opening an issue so it will be done in a separate PR at some point.